helix-core/src/main/java/org/apache/helix/controller/stages/TopStateHandoffReportStage.java

```diff
@@ -106,6 +107,7 @@ private void updateTopStateStatus(ResourceControllerDataProvider cache,
       }
 
       String resourceName = resource.getResourceName();
+      ImmutableMap.Builder<String, Long> failingPartitionsInfoCollector = ImmutableMap.builder();
```

Comment:
Let's not recreate this object. We can just make it a private member for this class. Every pipeline we clear it at beginning instead recreate the object every time.

Reply:
Yes, thanks @junkaixue, thats a very good point. I have updated the code now.